static/app/views/settings/components/teamSelect.tsx

```diff
@@ -92,6 +93,23 @@ function TeamSelect({
   const {teams, onSearch, fetching} = useTeams();
   const {orgRoleList, teamRoleList} = organization;
 
+  const slugsToFilter: string[] =
+    selectedTeams?.map(tm => tm.slug) || selectedTeamRoles?.map(tm => tm.teamSlug) || [];
+
+  // Determine if adding a team changes the minimum team-role
+  // get teams with org roles, if any
+  const orgRolesFromTeams = teams
+    .filter(team => slugsToFilter.includes(team.slug) && team.orgRole)
+    .map(team => team.orgRole ?? '');
+
+  if (selectedOrgRole) {
+    orgRolesFromTeams.push(selectedOrgRole);
+  }
+
+  // Sort them and to get the highest priority role
+  // Highest prio role may change minimum team role
+  const topOrgRole = getTopOrgRole(orgRolesFromTeams, orgRoleList)?.id;
```

Comment:
I'm still not a fan of `topOrgRole` because it could mean 2 things:

1. The top-dog org-role
1. The highest org-role for a member

Reply:
I'd toss in a vote for effectiveOrgRole